### PR TITLE
Validation service allow list

### PR DESCRIPTION
**Pull request by CManole**
## Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [x] Make sure `mvn -P integration-test clean verify` runs for the whole project and, if you touched any code for a service in the `services` folder, ensure it can be run with `spring-boot:run`

### CODE REVIEW

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/DigitalGreenCertificateToCborMapping.java [hunk 1]
```diff
@@ -90,7 +90,7 @@ public DigitalGreenCertificateToCborMapping(DigitalCovidCertificateClient digita
   }
 
   /**
-   * CBOR encoding of {@code constructRules}.
+   * CBOR encoding of {@codeDccValidationAllowListSignatureTest.java constructRules}.
```
Comment (roesslerj) [inline services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/DigitalGreenCertificateToCborMapping.java:93]: I don't think that annotation exists 😆

**Review by roesslerj — CHANGES_REQUESTED** [review]

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/BusinessRule.java [hunk 2]
```diff
@@ -167,5 +167,4 @@ public void setDesc(String desc) {
     Invalidation,
     BoosterNotification
   }
-
```
Comment (ioangut) [inline services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/BusinessRule.java:170]: I guess this is a unwanted change in this class

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/dsc/DigitalCovidValidationCertificateToProtobufMapping.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+   * Validates an object (JSON) based on a provided schema containing validation rules.
+   * @param allowList - object to be validated
+   * @throws JsonProcessingException - if object to be validated fails on JSON processing
+   * @throws ValidationException - if the validation of the object based on validation schema fails.
+   */
+  public boolean validateSchema(AllowList allowList) {
+    try (final InputStream in = resourceLoader.getResource(DCC_VALIDATION_RULE_JSON_CLASSPATH).getInputStream()) {
+      validateJsonSchema(allowList, in);
+      return true;
+    } catch (ValidationException | IOException e) {
```
Comment (ioangut) [inline services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/dsc/DigitalCovidValidationCertificateToProtobufMapping.java:54]: ValidateJsonSchema method can also throw JsonProcessingException, please catch that too.

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/dsc/DigitalCovidValidationCertificateToProtobufMapping.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+      LOGGER.error(e.getMessage(), e);
```
Comment (ioangut) [inline services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/dsc/DigitalCovidValidationCertificateToProtobufMapping.java:55]: Maybe we can enhance this logging.

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/dsc/DigitalCovidValidationCertificateToProtobufMapping.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    try {
+      PublicKey publicKey = getPublicKeyFromString(
+          distributionServiceConfig.getDigitalGreenCertificate().getAllowListCertificate());
+      ecdsaSignatureVerification(
+          signature,
+          publicKey,
+          content.getBytes(StandardCharsets.UTF_8));
+      return true;
+    } catch (Exception e) {
+      LOGGER.error(e.getMessage(), e);
```
Comment (ioangut) [inline services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/dsc/DigitalCovidValidationCertificateToProtobufMapping.java:76]: same here, enhance the logging.

File: services/distribution/src/main/resources/application.yaml [hunk 6]
```diff
@@ -255,6 +258,7 @@ services:
           trust-store: ${DCC_TRUST_STORE:../../docker-compose-test-secrets/dcc_truststore}
           trust-store-password: ${DCC_TRUST_STORE_PASSWORD:123456}
 
+
```
Comment (ioangut) [inline services/distribution/src/main/resources/application.yaml:261]: new line not needed

**Review by ioangut — CHANGES_REQUESTED** [review]
Please see the above comments

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/DigitalGreenCertificateToCborMapping.java (hunk 1 above)
Comment (CManole) [reply services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/DigitalGreenCertificateToCborMapping.java]: fixed :)

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/dsc/DigitalCovidValidationCertificateToProtobufMapping.java (hunk 3 above)
Comment (CManole) [reply services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/dsc/DigitalCovidValidationCertificateToProtobufMapping.java:54]: yes, but it's a IOException, unless you want us to log that differently, it's no difference.

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/dsc/DigitalCovidValidationCertificateToProtobufMapping.java (hunk 4 above)
Comment (CManole) [reply services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/dsc/DigitalCovidValidationCertificateToProtobufMapping.java]: done

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/dsc/DigitalCovidValidationCertificateToProtobufMapping.java (hunk 5 above)
Comment (CManole) [reply services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/dsc/DigitalCovidValidationCertificateToProtobufMapping.java]: added reason for the error. if not good, let me know.

File: services/distribution/src/main/resources/application.yaml (hunk 6 above)
Comment (CManole) [reply services/distribution/src/main/resources/application.yaml:261]: removed

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/BusinessRule.java (hunk 2 above)
Comment (CManole) [reply services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/BusinessRule.java:170]: there was an extra space. when i added a new RuleType i made changes, but then i saw it's not necessary. if you want the extra space, i can roll it back :)

File: services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/dsc/DigitalCovidValidationCertificateToProtobufMapping.java [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+   * @param allowList - object to be validated
+   * @throws JsonProcessingException - if object to be validated fails on JSON processing
+   * @throws ValidationException - if the validation of the object based on validation schema fails.
+   */
+  public boolean validateSchema(AllowList allowList) {
+    try (final InputStream in = resourceLoader.getResource(DCC_VALIDATION_RULE_JSON_CLASSPATH).getInputStream()) {
+      validateJsonSchema(allowList, in);
+      return true;
+    } catch (ValidationException e) {
+      LOGGER.error("Schema invalid", e);
```
Comment (ioangut) [inline services/distribution/src/main/java/app/coronawarn/server/services/distribution/dgc/dsc/DigitalCovidValidationCertificateToProtobufMapping.java:55]:
```suggestion
      LOGGER.error("Json schema validation failed", e);
```

**Review by ioangut — APPROVED** [review]

**Review by roesslerj — APPROVED** [review]

**Pull request merged**